.travis.yml

```
@@ -1,17 +1,37 @@
-  - secure: IHD6Qj30A6eN/0cslyeMTK4iiJwZRiMBEbRqwyVfZdfhF4yIuwO6MQA1n6uCybstR+qRY1Wwmb7HCU2C7RnlVf1CU71FYOxDkWaymEeBWTK97W5xqZPb35i+fm/WKiQo4wyaRIFJZ8XYX5W9+jf3+n9qo6p6MU/SmT6GHgcb3rmfFjS7tiiy2vVZURoI4LT2+q3nfBRBrvdLrFNJf/DisIQuN1aYzM8xflB7EMCjuZS/4X8A/yktl1GNe2WqM3wT2XNuBFPVMDTyInYVYlSN3ZnX8sxzBSwPMQmz6iNk+6Fb3JJcZ7PIZPah3kC9yr5MS5zCjeK5ozyOX/pABlHEjnybw3733jN2OtdU1bRwQzyTHJGA4oz9MG0jbONl6vndQujP9NW+MFZNF/cyPN/dO3wTXylcCEwROkfffjZRSExo2S7mVL3oaczL4nOsdulZ4cgPDxC0w/PSprgkhYZaNRH3nJ12VTFFoc1TlVnkyo7OYFkb82StiD2yLmA9YProidnykeAEk68Pw78OfNU4mQF/9mFBuntWNikAhNwGdw/Rj+j3jywx+AP6Ik/NWLKU2oLWzRN9q5YxeL05myeECDdnp9QMSDrUsYte2ojojORbLJNzpHY2uDazObmj6i86XxVZ5fZrjbYWWRL8HsOslB7IJSApHGEHpsiTFgcpsUE=
-  - secure: iU4T1WHEH3xZPuJWbPLEq/t/bfWoKhaujAQcTIkSEG63TJX5eTrLZZy8IEqwzPHasDUMZPjHrAHSJx6tiXRCwdITEzICBySDLNQfLGX7h2X5OSFj/r9R9zWkwbihsG9N9CZ8+jX2btiUH01l0sqM0SoKnlfLqBF6oeYsNinyHNgOmPDtZjdDflQK8Saoos0yE3uwExfGTVxjK36zox8qyK3I5hGGmLE91fQP/hKov88H80qrcdsjOi35nktN2G0CdXEdGmXGzkT+iJ9UFXjB0X85Q3lLrvF1Rli0toPpHnQXNDeRmW+EEE4tnjUL02qTQtV/lfVMqD4R26I58U4HGgv1WebKYyIwAwTJ6xqoR+qtw3KXYuevMqtH0SqihO6T1/J09paC5sAea+zxwRgFCt0/k8CkG3jNayeoGcj26+owMLibMefD1V12miENfxrCjqcWH5T8TiynkKaqRtjfZkwSuqFxteO16U/vAx/DZLJ/Q4kUk5X3b9T+ScxRDzEYL0VRfmgL836weBwR5GHY1fKv4wHTgFUAf9Wwm/Y8VUhKOGoyyYHwpJ+/kS6IKMJoeHm6CNeEvjUBaz+wZ8UrPPMJZ2MQE9rhyEOFwHIHPyVbMpkpxNUHn2mhNuS7S7txCAVl/IWKEE8KFx8W/vfW1WcbkCfeFS8m2/FN0zPiy2M=
+after_success:
+- echo $DOCKER_PASSWD | docker login --username $DOCKER_USER --password-stdin
+- docker pull $DOCKER_USER/todo-app:latest
+- docker build --cache-from $DOCKER_USER/todo-app:latest --target production --tag
+  $DOCKER_USER/todo-app:latest .
+- docker tag $DOCKER_USER/todo-app:latest $DOCKER_USER/todo-app:$TRAVIS_COMMIT
+- docker push $DOCKER_USER/todo-app:latest
+- docker push $DOCKER_USER/todo-app:$TRAVIS_COMMIT
+- docker tag $DOCKER_USER/todo_app registry.heroku.com/$HEROKU_APP/web
+deploy:
+  provider: heroku
+  api_key:
+    secure: gHIDdLrI4zE6xVLWF+dpevj6wixO3YrWVo3T22VKQU/Qk48+erowXL1mmP658X0+GI5uLKDpvkTs97so+28+yGl0NIQwC+1FImmnqVoKrQhHoJVKjCGSetCZp1c71E/c0Tpmmd6ApeuPDs6UKg7rnNLhNpTH7INjch3OtqymxS8qS3mFULehF936PpLyzlsWc6aO0KpUrudXgTFFUDA4FAkndpI22bA2AiaPAZkBvIpHJ+qZnLrpH7/DkwlCPPDIttm9yO0sFymR1s/55z0A/W1ZLXfMCRUobL0MaCAjTJ8/0YQvnK583BsQXSFUPwQGVYI5uUIm9/4Zoprgx7ZN27bEHIjEB6Meq3PUUCn05zbC1npfKa6TC0+xgx8Ij0lhlZko6tnvx42kQvGctMUM5SCW+m5OpKiVZ8rpqdEjUg6iNFGiciyZVPpSckKWTg30GwwDbrsU2hWfbCZxmXSgx9kVI5fDJ/uH0Jeh+HpcffqBUgg47rViaZmbwG1vMmc2c11k+o4xrVeoSei7IYVSC/NUN4fasqix2dv/q2KGPwUQdNMqST8zPkIpnaamCkAx3IB/nQ+gFcfdzwCG193NIsDej3fAJ00BXbRgg4nqFzSZqZUSJWEL/33QtWBjc3hoBL7xasxDXqjzAJAEM7qrPgiGikO44LGNMPPJOG2Agps=
+  app: todo-las
```

I know you're not finished with M8, but worth noting we're expecting to deploy manually with the `script` provider rather than with the built in Heroku provider - not sure if that might be causing some of your headaches!
E.g. in Travis you might have:
```suggestion
  provider: script
  script: bash scripts/deploy.sh
```

With your `deploy.sh` script:
* Pushing to the Heroku image repo (will require logging in)
* Triggering a deployment

---

If you're still having any issues feel free to ask for more advice!